File: Stacks/StockSpan/span.py
class Solution:
    def calculateSpan(self, arr):
        # code here
        n = len(arr)
        ans = [1]*n
        st = []
        st.append(0)
        for i in range(1,n):
            while len(st) and arr[i]>=arr[st[-1]]: st.pop()
            if len(st): ans[i]=i-st[-1]
            else: ans[i] = i+1
            st.append(i)
        return ans

File: Stacks/StockSpan/span.cpp
#include <vector>
#include <stack>
using namespace std;

class Solution {
  public:
    vector<int> calculateSpan(vector<int>& arr) {
        // code here
        int n = arr.size();
        vector<int>ans(n,1);
        stack<int>st;
        st.push(0);
        for(int i=1;i<n;i++){
            while(st.size() && arr[i]>=arr[st.top()])st.pop();
            if(st.size())ans[i]=i-st.top();
            else ans[i] = i+1;
            st.push(i);
        }
        return ans;
    }
};